app/models/v2/report.rb

```diff
@@ -44,12 +86,25 @@ def percent_compliant
     has_many :report_systems, class_name: 'V2::ReportSystem', dependent: nil # rubocop:disable Rails/InverseOf
     has_many :systems, class_name: 'V2::System', through: :report_systems
     has_many :reported_systems, class_name: 'V2::System', through: :test_results, source: :system, dependent: nil
+    # rubocop:disable Rails/InverseOf
+    has_many :reporting_and_non_reporting_systems, lambda {
+      joins(
+        arel_table.join(V2::System.arel_table, AN::InnerJoin)
+                  .on(V2::System.arel_table[:id].eq(V2::ReportSystem.arel_table[:system_id]))
+                  .join(V2::TestResult.arel_table, AN::OuterJoin)
+                  .on(V2::TestResult.arel_table[:system_id].eq(V2::System.arel_table[:id]))
+                  .join_sources
+      ).where(
+        Pundit.policy_scope(User.current, V2::System).where_clause.ast.to_sql
```

Comment:
If you don't utilize a `where` clause over the explicitly joined `TestResult` model, it will be omitted by ActiveRecord. In my opinion this is a bug and will take a look at how this could be fixed in the future. This policy scoping is unnecessary, you just need ANY `where` condition involving the `TestResult` model to make this work instead. 

Be careful that you can't use `where.not(test_results: { id: nil })` as it could render the `LEFT OUTER JOIN` useless, instead you can create a complex arel-style (OR is hard in pure ActiveRecord) condition that allows both `NULL` AND `NOT NULL` for the field.